File: org.boblycat.frida.core/src/java/org/boblycat/frida/core/PrintUtils.java
package org.boblycat.frida.core;

/**
 *
 * Created: Feb 26, 2010
 *
 * Part of Frida IRE.
 * Copyright (c) 2010, Karl Trygve Kalleberg, Ole André Vadla Ravnås
 * Licensed under the GNU General Public License, v3
 *
 *
 * @author: karltk@boblycat.org
 */

public class PrintUtils {

	private static String toBinaryString(int b, int bits) {
		int v = 1 << (bits - 1);
		String s = "";
		for(int i = 0; i < bits; i++) {
			if((b & v) == v) {
				s += "1";
			} else { 
				s += "0";
			}
			v >>= 1;
		}
		return s;
	}
	
	public static String toBinaryString(int b) {
		return toBinaryString(b, 32);
	}
	
	public static String toBinaryString(short b) {
		return toBinaryString(b, 16);
	}
	
	public static String toBinaryString(byte b) {
		return toBinaryString(b, 8);
	}

	public static String toPaddedBinaryString(byte b) {
		return String.format("%08s", toBinaryString(b)); 
	}
}

File: org.boblycat.frida.core/src/org/boblycat/frida/core/PrintUtils.scala
package org.boblycat.frida.core

object PrintUtils {

	private def toBinaryString(b : Int, bits : Int) : String = {
		var v = 1 << (bits - 1)
		var s = ""
		for(i <- 0 until bits) {
			if((b.toInt & v) == v) { s += "1" } else { s += "0" }
			v >>= 1
		}
		s
	}
	
	def toBinaryString(b : Int) : String = toBinaryString(b, 32)
	def toBinaryString(b : Short) : String = toBinaryString(b, 16)
	def toBinaryString(b : Byte) : String = toBinaryString(b, 8)

	def toPaddedBinaryString(b : Byte) = {
		val s = b.toBinaryString
		val miss = 8 - s.length
		"0" * miss + s 
	}
}
